src/clj_okhttp/formats.clj

```diff
@@ -35,29 +37,40 @@
                                (okhttp/->request-body content-type (encoder body (.-charset negotiated)))
                                (throw (ex-info (str "No muuntaja encoder defined for " content-type) {}))))))]
     (cond-> request
-      (some? request-body) (assoc :body request-body)
-      (some? content-type) (assoc-in [:headers "content-type"] content-type))))
-
-(defn resolve-content-type [content-type]
-  (case content-type
-    :stream "application/octet-stream"
-    :csv "text/csv"
-    :json "application/json"
-    :yaml "application/yaml"
-    :edn "application/edn"
-    :clojure "application/edn"
-    :transit "application/transit+json"
-    (or content-type "application/octet-stream")))
-
-(defn format-stream [muuntaja content-type stream]
-  (let [resolved   (resolve-content-type content-type)
-        negotiated ^FormatAndCharset (negotiate-content-type muuntaja resolved)]
-    (if-some [decoder (m/decoder muuntaja (.-format negotiated))]
-      (decoder stream (.-charset negotiated))
-      stream)))
+            (some? request-body) (assoc :body request-body)
+            (some? content-type) (assoc-in [:headers "content-type"] content-type))))
+
+(def coercion-mapping
+  {:csv     "text/csv"
+   :json    "application/json"
+   :yaml    "application/yaml"
+   :edn     "application/edn"
+   :clojure "application/edn"
+   :transit "application/transit+json"})
+
+(defn coerce-stream [^InputStream stream coercion detected-charset]
+  (case coercion
+    :text (slurp stream :encoding detected-charset)
+    :stream stream
+    :byte-array (with-open [in  stream
```

Comment:
In my experience, too often developers carelessly stuff data into byte arrays when they could've handled it in a streaming fashion instead. I guess I purposefully omitted this to force them to reckon with a stream.

Reply:
@RutledgePaulV , it's often about convenience. Dealing with a stream can often be a lot more cumbersome especially if the first thing you plan on doing is reading it into a byte array. Making sure things are properly consumed, opened/closed, etc can be tricky for those less experienced. IMO it's no different than trying to say "please coerce this to a clj map please" instead of asking a developer to use something like Cheshire's json streaming. I understand your point, though I don't see a reason to not provide this.